crates/runtime/src/module_id.rs

```diff
@@ -7,6 +7,13 @@ use std::sync::atomic::{AtomicU64, Ordering};
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct CompiledModuleId(u64);
 
+impl CompiledModuleId {
+    /// A module ID denoting "no module".
+    pub fn none() -> Self {
+        Self(u64::MAX)
+    }
+}
```

Comment:
Why not change `CompiledModuleId` to be a newtype around `NonZeroU64` and then we can use plain `Option` and `None` without pessimizing size, instead of having our own custom sentinel that we can forget to check for?

Reply:
Also, this is an existing issue (introduced in the memfd PR?) that is maybe going to become worse with these changes, but we use a relaxed ordering in the `fetch_add` when allocating new `CompiledModuleId`s below -- it seems like this could be a problem on non-x86?

Reply:
Re: relaxed ordering, that's fine -- all that means is that the atomic ID-allocation event is not synchronized with respect to any other loads/stores in the system (there are no fences). But the fetch-add itself is still an atomic action, with some linear ordering amongst the other accesses to the "next ID" atomic. That's sufficient for the invariants we need: we just need unique IDs for each module, we don't care what they are.

(Good idea wrt `NonZeroU64`; I'll make that change!)

Reply:
Ah yes, I didn't think about how the `fetch_add` itself was still atomic. Thanks!